astroid/modutils.py

```diff
@@ -154,11 +154,7 @@ class NoSourceFile(Exception):
 
 
 def _normalize_path(path):
-    return os.path.normcase(os.path.abspath(path))
-
-
-def _canonicalize_path(path):
-    return os.path.realpath(os.path.expanduser(path))
+    return os.path.normcase(os.path.realpath(os.path.expanduser(path)))
```

Comment:
Should we also use ``os.path.expandvars()`` ?

Reply:
As far as I know, `path` is always absolute and does not contain variables. Actually, all the tests pass even without `os.path.expanduser()` on both macOS and Linux (w/ and w/o venv). So maybe we can get rid of both `expandvars()`and `expanduser()`? 

I would appreciate confirmation from others as it might depend on the OS or the way Python is installed.

Reply:
I think it depends on the way ``_canonicalize_path`` is used. This look like a very generic function that someone could start calling in an unanticipated way but right now it's at least private and used only in ``modpath_from_file``. So here https://github.com/PyCQA/astroid/blob/main/astroid/builder.py#L135 and here https://github.com/PyCQA/astroid/blob/main/astroid/manager.py#L112. Guess the filepath at this point should not have any user vars in it. If it does wit should be handled before.

Reply:
Yes right now it's not a problem. How about we add a comment to remind that variables and `~` need to be expanded beforehand in case someone is going to use it in other places?

Reply:
Sounds good to me !